app/src/main/java/com/gh4a/activities/RepositoryActivity.java

```diff
@@ -182,24 +125,46 @@ protected void onResultReady(Boolean result) {
     private ContentListContainerFragment mContentListFragment;
     private CommitListFragment mCommitListFragment;
     private RepositoryEventListFragment mActivityFragment;
+    private Gh4Application mApp;
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        mApp = (Gh4Application) getApplicationContext();
 
         mActionBar = getSupportActionBar();
         mActionBar.setTitle(mRepoOwner + "/" + mRepoName);
         mActionBar.setDisplayHomeAsUpEnabled(true);
 
         setContentShown(false);
 
-        getSupportLoaderManager().initLoader(LOADER_REPO, null, mRepoCallback);
+
+        loadRepository(false);
+
         if (Gh4Application.get().isAuthorized()) {
-            getSupportLoaderManager().initLoader(LOADER_WATCHING, null, mWatchCallback);
-            getSupportLoaderManager().initLoader(LOADER_STARRING, null, mStarCallback);
+            isWatching(false);
+            isStarring(false);
         }
     }
 
+    public void loadRepository(boolean refresh) {
+        Consumer subscriber = repository -> {
+            mRepository = (Repository) repository;
+            updateTitle();
+            invalidateTabs();
+            // Apply initial page selection first time the repo is loaded
+            if (mInitialPage >= PAGE_REPO_OVERVIEW && mInitialPage <= PAGE_ACTIVITY) {
+                getPager().setCurrentItem(mInitialPage);
+                mInitialPage = -1;
+            }
+            setContentShown(true);
+            supportInvalidateOptionsMenu();
+        };
+
+        RepositoryService.loadRepository(this, mRepoOwner, mRepoName, refresh)
+                .subscribe(subscriber, e -> {});
```

Comment:
If there is no matter error in the chain, it will be propagated to every next error handler in the chain until the error handler within the `subscribe` method

Reply:
Yes, but what does this have to do with removing an empty handler?

Reply:
if I m not wrong this was resulting in an **ErrorHandlerNotImplemented** or something **Exception**, causing the app to crash

Reply:
Given there's [a variant of subscribe() without error handler parameter](http://reactivex.io/RxJava/javadoc/io/reactivex/Observable.html#subscribe-io.reactivex.functions.Consumer-) I have issues believing this is unavoidable ;-)

Reply:
That's what I thought too, but looking at the method code from RxJava library 

```
    /**
     * Runs the source observable to a terminal event, ignoring any values and rethrowing any exception.
     * <dl>
     *  <dt><b>Scheduler:</b></dt>
     *  <dd>{@code blockingSubscribe} does not operate by default on a particular {@link Scheduler}.</dd>
     * </dl>
     * @since 2.0
     */
    @SchedulerSupport(SchedulerSupport.NONE)
    public final void blockingSubscribe() {
        ObservableBlockingSubscribe.subscribe(this);
    }
```